app/utils/queryBuilder.ts

```diff
@@ -317,9 +196,13 @@ const getWhereExp = (filterGroups: FilterGroup[]): string => {
  *    }
  *  }
  */
-const buildQuery = (
-  {
-    columns,
+const buildQuery = <T extends Record<string, unknown>>(
+  queryConfig: QueryConfig,
+  columns: ColDataCardDef<T>[],
+  includeAggregates: boolean,
+  contextFilters?: Filter[]
+): string => {
```

Comment:
I need typescript 101 help here, what does adding `<T extends Record<string, unknown>>` add to our type?

Reply:
because we added `ColDataCardDef<T>` as a prop, and it requires a generic (`T`), we need to supply `T` in our function declaration. that's just the syntax for using generics.

generics themselves can be of a specific base type, and in our case the type for our `ColDataCardDef<T>` is  an extension of the `Record<string, unknown>` utility type. You can see that this specific type comes from our definition of  the `ColDataCardDef` interface in `types.ts.` 

So, to be compliant with the generic type `ColDataCardDef` is expecting, we need specify that this function specifically excepts `<T extends Record<string, unknown>>` rather than plain 'ole `T`.

I hope that helps! If you remove `extends Record<string, unknown>` from this function you will see the way typescript barks about it.

Reply:
this does help! thank you